### DISCUSSION TITLE
增加字体库选项，使用工作目录中的字体进行编译

### DISCUSSION THREAD

**Discussion by CalaW** [tuna/thuthesis#910] [Ideas]
能否增加一个选项`fontset=local`，使用当前工作目录下`fonts`文件夹里的字体进行编译？这样就解决了类似 #747 中的问题，同时不需要手动修改 .cls 文件中对应内容。

可行的话我可以写个 PR

**Comment by Harry-Chen** [comment]
我个人倾向于不做处理，如 #747 所说可以使用 TeXPage 等平台，我个人更推荐在最后使用本地编译

**Comment by zepinglee** [comment]
> 能否增加一个选项`fontset=local`，使用当前工作目录下`fonts`文件夹里的字体进行编译？这样就解决了类似 #747 中的问题，同时不需要手动修改 .cls 文件中对应内容。
> 
> 可行的话我可以写个 PR

`local` 这个名称不太合适，因为 `fontset` 的含义是将宋体、黑体等分别对应到具体的字体/文件名。

我提议在 `fontset=windows` 时增加一个分支 `\IfFileExists{\thu@font@dir/SimSun.ttf}{}{}`，类似 `\thu@mac@word@font@path` 的情况。

https://github.com/tuna/thuthesis/blob/93a582f24d232fcbdf802cc82fe955cd65068732/thuthesis.dtx#L2544-L2553

**Comment by zepinglee** [comment]
在 eb938d8 中实现。

eb938d8 修改为，若系统不存在 SimSun 字体，会尝试调用 Windows 字体的文件（`Simsun.ttc`, `SimHei.ttf`, `Kaiti.ttf`, `Fangsong.ttf`）。默认从当前目录寻找，也可以在 `\documentclass` 参数中使用 `windows-font-dir` 设置字体的目录，例如字体在 `fonts/` 目录下，则设置 `\documentclass[degree=master, windows-font-dir=fonts]{thuthesis}`。
